common/src/common/messages.py

```diff
@@ -106,14 +107,50 @@ def __init__(self, receiver: str) -> None:
         self.payload = {"magic": CANS_PEER_HANDSHAKE_MAGIC}
 
 
+class AckMessageDelivered(CansMessage):
+    """Message delivery acknowledgement."""
+
+    def __init__(self, receiver: str, cookie: str) -> None:
+        """Create a delivery acknowledgement."""
+        super().__init__()
+        self.header.msg_id = CansMsgId.ACK_MESSAGE_DELIVERED
+        self.header.receiver = receiver
+        self.payload = {
+            "cookie": cookie,
+        }
+
+
+class NackMessageNotDelivered(CansMessage):
+    """Message delivery failed notification."""
+
+    def __init__(
+        self,
+        receiver: str,
+        message_target: str,
+        message_id: CansMsgId,
+        extra: str,
+        reason: str,
+    ) -> None:
+        """Create a delivery failure notification."""
+        super().__init__()
+        self.header.msg_id = CansMsgId.NACK_MESSAGE_NOT_DELIVERED
+        self.header.receiver = receiver
+        self.payload = {
+            "message_target": message_target,
+            "msg_id": message_id,
+            "extra": extra,
+            "reason": reason,
+        }
+
+
 class SchnorrCommit(CansMessage):
     """Commitment of the Schnorr identification scheme."""
 
     def __init__(self, public_key: str, commitment: str) -> None:
         """Create a Schnorr commitment message."""
         super().__init__()
         self.header.msg_id = CansMsgId.SCHNORR_COMMIT
-        self.header.receiver = ""
+        self.header.receiver = None
```

Comment:
receiver by default is None, right? Or do you want to emphasize that? :)

Reply:
Just to emphasize :)